--- os_dep/linux/rtw_android.c	2014-05-05 10:28:42.000000000 +0200
+++ os_dep/linux/rtw_android.c	2016-02-06 12:18:09.510889002 +0100
@@ -337,7 +337,7 @@
 {
 	int cmd_num;
 	for(cmd_num=0 ; cmd_num<ANDROID_WIFI_CMD_MAX; cmd_num++)
-		if(0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
+		if(0 == strncasecmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
 			break;
 
 	return cmd_num;
