--- os_dep/linux/ioctl_linux.c	2014-05-05 10:28:42.000000000 +0200
+++ os_dep/linux/ioctl_linux.c	2016-02-08 10:36:19.000000000 +0100
@@ -580,11 +580,27 @@
 	//parsing WPA/WPA2 IE
 	if (pnetwork->network.Reserved[0] != 2) // Probe Request
 	{
-		u8 buf[MAX_WPA_IE_LEN*2];
-		u8 wpa_ie[255],rsn_ie[255];
+		u8 *buf;
+		u8 *wpa_ie, *rsn_ie;
 		u16 wpa_len=0,rsn_len=0;
 		u8 *p;
 		sint out_len=0;
+
+		buf = kzalloc(MAX_WPA_IE_LEN*2, GFP_ATOMIC);
+		if (!buf)
+			goto exit;
+		wpa_ie = kzalloc(255, GFP_ATOMIC);
+		if (!wpa_ie) {
+			kfree(buf);
+			goto exit;
+		}
+		rsn_ie = kzalloc(255, GFP_ATOMIC);
+		if (!rsn_ie) {
+			kfree(buf);
+			kfree(wpa_ie);
+			goto exit;
+		}
+
 		out_len=rtw_get_sec_ie(pnetwork->network.IEs ,pnetwork->network.IELength,rsn_ie,&rsn_len,wpa_ie,&wpa_len);
 		RT_TRACE(_module_rtl871x_mlme_c_,_drv_info_,("rtw_wx_get_scan: ssid=%s\n",pnetwork->network.Ssid.Ssid));
 		RT_TRACE(_module_rtl871x_mlme_c_,_drv_info_,("rtw_wx_get_scan: wpa_len=%d rsn_len=%d\n",wpa_len,rsn_len));
@@ -635,6 +651,10 @@
 			iwe.u.data.length = rsn_len;
 			start = iwe_stream_add_point(info, start, stop, &iwe, rsn_ie);		
 		}
+
+		kfree(buf);
+		kfree(wpa_ie);
+		kfree(rsn_ie);
 	}
 
 	{ //parsing WPS IE
@@ -766,7 +786,8 @@
 		iwe.u.data.length = strlen(buf);
 		start = iwe_stream_add_point(info, start, stop, &iwe, buf);
 	}
-	
+
+exit:
 	return start;	
 }
 
@@ -10626,7 +10647,7 @@
 		GetThermalMeter(padapter, &thermal);
 		printk("%s ### REG_C80:0x%08x,REG_C88:0x%08x ,thermal:0x%02x####\n",__FUNCTION__,
 			rtw_read32(padapter,0xc80),rtw_read32(padapter,0xc88),thermal);
-	}	
+	}	
 	return 0;
 }
 
